src/currency.rs

```diff
@@ -0,0 +1,51 @@
+use core::fmt;
+use num_derive::FromPrimitive;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+#[cfg(feature = "build-binary")]
+use clap::ValueEnum;
+
+#[derive(FromPrimitive, Debug, Copy, Clone, Hash, Serialize, Deserialize, PartialEq, Eq)]
+#[cfg_attr(feature = "build-binary", derive(ValueEnum))]
+pub enum Currency {
+    Arweave = 1,
+    Solana = 2,
+    Ethereum = 3,
+    Erc20 = 4,
+    Cosmos = 5,
+}
```

Comment:
Mapping enums to integers is often a bad idea and can cause things go wrong if someone stores these values somewhere. Is this really needed, I wasn't able to see where?

Reply:
What should be a better approach to integers?